test/provisioner/provision_aws.go

```diff
@@ -340,6 +345,55 @@ func (v *Vpc) createSubnet() error {
 	return nil
 }
 
+func (v *Vpc) createSecondarySubnet() error {
+	if v.SecondarySubnetId != "" {
+		return nil
+	}
+
+	// EKS requires at least two subnets and they should be on different
+	// Availability zones on the same region. So let's ensure this secondary
+	// subnet's AZ don't clash with the primary's.
+	subnets, err := v.Client.DescribeSubnets(context.TODO(),
```

Comment:
Have you seen some documentation on this requirement? it seems to be a common practice for managed clusters also in Azure cc @esposem

Reply:
Interesting, I didn't know Azure had such as requirement too.

In https://docs.aws.amazon.com/eks/latest/userguide/network_reqs.html you gonna read `When you create a cluster, you specify a [VPC](https://docs.aws.amazon.com/vpc/latest/userguide/configure-your-vpc.html) and at least two subnets that are in different Availability Zones`.

I tried to create with one subnet and with two on same AZ but the API denied on both cases.